docs/DeveloperGuide.md

```diff
@@ -376,6 +376,127 @@ Step 14: Lastly, `EditCommand` creates a `CommandResult` with `SuccessMessage` a
 
 _{more aspects and alternatives to be added}_
 
+### 6. Remark feature
+
+`[written by: Low Ming Lim]`
+
+The remark feature allows users to add a custom note to a new or existing patient using the `r/` tag in multiple commands.
+This provides our users with the flexibility and freedom to store extra notes or bio data for a patient apart from the compulsory 
+fields such as name, phone number, etc.
+
+#### 6.1 Implementation
+This feature creates a `Remark` instance which is stored internally in Nuudle as a variable of a `Patient` object
+which is in turn stored in the `PatientBook`. These classes are a part of the `Model` component and are illustrated
+in the class diagram below. 
+
+![RemarkLogicClassDiagram](images/RemarkLogicClassDiagram.png)
+<br>**Diagram 6.1.1: Class diagram for classes involved in the remark feature of the Model component**
+ 
+Additionally, to facilitate greater convenience for our users, we have implemented our remark feature to support the following pathways:
+
+1. Adding a remark via the `AddCommand` for a new `Patient`: 
+    
+    Command syntax: `add n/NAME i/NRIC p/PHONE_NUMBER a/ADDRESS [r/REMARK] [t/TAG]…​`
+    
+    Example Usage:
+    * `add n/Betsy i/S9123456G t/friend a/NUS Utown p/1234567 r/Prefers Dr John`
+    * `add n/John Doe i/S9730284G p/98765432 a/John street, block 123, #01-01 r/Regular customer`
+
+1. Adding a remark via the `RemarkCommand` for an existing `Patient`:
+    
+    Command syntax: `remark INDEX [r/REMARK]`
+    
+    Example Usage:
+    * `remark 2 r/Has been visiting Dr John`
+    * `remark 1 r/Can only converse in mandarin`
+
+1.  Or via the `EditCommand`:
+ 
+    Command syntax: `edit INDEX [n/NAME] [i/NRIC] [p/PHONE_NUMBER] [a/ADDRESS] [r/REMARK] [t/TAG]…`
+    
+    Example Usage:
+    * `edit 1 r/Can only converse in mandarin`
+    * `edit 2 n/Betsy r/Has been visiting Dr John`
+
+Refer to the following activity diagram for a summary of the above pathways.
+![RemarkPathwaysActivityDiagram](images/RemarkPathwaysActivityDiagram.png)
+<br>**Diagram 6.1.2: Architecture diagram showcasing available pathways to create a remark**
+
+This segment will focus on the implementation details for the `RemarkCommand` pathway. The implementation for the 
+alternative `AddCommand` and `EditCommand` pathways can be found in another segment of our Developer's Guide.
+
+ 
+The addition of a remark via the remark command pathway is mainly facilitated by the `RemarkCommand` class which
+extends the abstract class `Command`. A `RemarkCommandParser` which implements the `Parser` interface is required 
+to instantiate a `RemarkCommand` from the user input. The classes mentioned above in this paragraph resides in our 
+`logic` component.
+
+#### 6.2 Implementation Illustration
+<br>![RemarkSequenceDiagram](images/RemarkSequenceDiagram.png)
+<br>**Diagram 6.2.1: Remark Command sequence diagram**
+
```

Comment:
Would `Sequence diagram of remark command` be better?

Reply:
Good idea!